samples/cpp/OpenCVsamples_cpp.h.in

```diff
@@ -0,0 +1,2 @@
+// the configured options and settings for Tutorial
+#define PATH_DATA_SAMPLE_OPENCV "@PROJECT_SOURCE_DIR@/samples/data"
```

Comment:
It is not very good to store absolute paths in binaries. This aproach works fine with binaries compiled by user, but it will not work for movable distribution packages.
Some "find"-like approach can be added instead:
1) check file directly (`lena.jpg`) and returns it if exists
2) check simple relative paths: ../data, ../samples/data, etc (`../data/lena.jpg`)
3) check `<CMake relative path between data and binary>/lena.jpg` (locations of build binaries and location of install paths are different)
4) something else, include absolute path. But specify absolute path via CMake add_definitions() and use it under `#ifdef` (I'm not sure about spaces in path). This will allow to remove generated files.
I believe that 1,2 and absolute path (4) will cover the most part of use cases.